client/components/data/document-head/index.jsx

```diff
@@ -33,19 +47,16 @@ class DocumentHead extends Component {
 		} );
 	}
 
+	componentDidMount() {
+		const { formattedTitle } = this.props;
+		document.title = formattedTitle;
+	}
+
 	componentWillReceiveProps( nextProps ) {
 		if ( this.props.title !== nextProps.title ) {
 			this.props.setTitle( nextProps.title );
```

Comment:
Nitpicking, but the component won't allow `undefined` title during initial mount, but will allow it upon subsequent re-renders. Alternatively consider `in` or `hasOwnProperty` in the will mount, or guard against `undefined` here.

```js
componentWillMount() {
	if ( 'title' in this.props ) {
		this.props.setTitle( this.props.title );
	}

	// ...
```